src/lein_monolith/plugin.clj

```diff
@@ -129,26 +150,29 @@
       (activate-profile profile-key)))
 
 
-
 ;; ## Plugin Middleware
 
 (defn middleware
   "Handles inherited properties in monolith subprojects by looking for the
   `:monolith/inherit` key."
-  [project]
-  (if (:monolith/inherit project)
-    ; Monolith subproject, add inherited profile.
-    (if (or (profile-active? project :monolith/inherited)
-            (profile-active? project :monolith/leaky))
-      ; Already activated, return project.
-      (do (lein/debug "One or both inherited profiles are already active!")
-          project)
-      ; Find monolith metaproject and generate profile.
-      (let [monolith (config/find-monolith! project)
-            profiles (build-inherited-profiles monolith project)]
-        (reduce-kv add-active-profile project profiles)))
-    ; Normal project, don't activate.
-    project))
+  ([project]
+   (middleware project nil))
+  ([project monolith]
```

Comment:
Interesting - for testing, I presume?

Reply:
It's an optimization used when building the merged profile, which internally calls this middleware function to insert the inherited profiles. That function already has a reference to the monolith, so there's no reason to read it each time it inserts the profiles.

Reply:
This leads me to thinking that the `each` task could be optimized a bit by passing in the monolith once instead of reading it per subproject.

Reply:
Ah, that makes sense. Injecting the inherited profiles from the monolith into the each subprojects was the optimization that originally caused #29 but maybe there's a way to do it while still using the middleware.